app/code/Magento/Downloadable/Model/Link/UpdateHandler.php

```diff
@@ -44,16 +44,18 @@ public function execute($entity, $arguments = [])
         $links = $entity->getExtensionAttributes()->getDownloadableProductLinks() ?: [];
         $updatedLinks = [];
         $oldLinks = $this->linkRepository->getList($entity->getSku());
-        foreach ($links as $link) {
-            if ($link->getId()) {
-                $updatedLinks[$link->getId()] = true;
+        if (!empty($links)) {
```

Comment:
It makes sense to move this condition one line upper to not get list of existing links